vaccine/healthcheck_ussd.py

```diff
@@ -260,12 +270,60 @@ async def state_terms(self):
                 "Reply"
             ),
             choices=[
-                Choice(next_state, self._("YES")),
+                Choice("state_privacy_policy", self._("YES")),
                 Choice("state_end", self._("NO")),
                 Choice("state_more_info_pg1", self._("MORE INFO")),
             ],
         )
 
+    async def state_privacy_policy(self):
+        if self.user.answers.get("confirmed_contact") == "yes":
+            next_state = "state_fever"
+        else:
+            next_state = "state_province"
+        if self.user.answers.get("state_privacy_policy_accepted"):
+            return await self.go_to_state(next_state)
+
+        if (
+            config.RAPIDPRO_URL
+            and config.RAPIDPRO_TOKEN
+            and config.RAPIDPRO_PRIVACY_POLICY_SMS_FLOW
+        ):
+            async with get_rapidpro() as session:
+                for i in range(3):
+                    try:
+                        data = {
+                            "flow": config.RAPIDPRO_PRIVACY_POLICY_SMS_FLOW,
+                            "urns": [f"tel:{self.inbound.from_addr}"],
+                        }
+                        response = await session.post(
+                            urljoin(config.RAPIDPRO_URL, "/api/v2/flow_starts.json"),
+                            json=data,
+                        )
+                        response.raise_for_status()
+                        break
+                    except HTTP_EXCEPTIONS as e:
+                        if i == 2:
+                            logger.exception(e)
+                            return await self.go_to_state("state_error")
+                        else:
+                            continue
```

Comment:
`state_privacy_policy` will be called twice, once to give the user the choices, and again when we receive the response to validate the user's choice, which would cause 2 smses to be sent.

We can avoid this by having a middle state, that sends the SMS, and then goes to this state without asking anything from the user.

Reply:
ah good catch thanks